text/profiles/0237-profiles-data-model.md

```diff
@@ -0,0 +1,1067 @@
+  repeated uint64 timestamps = 14;
+}
+
+// SymbolFidelity enumerates the level of fidelity for symbol information in a profile.
+enum SymbolFidelity {
+  SYMBOL_FIDELITY_UNSPECIFIED = 0; // The symbol fidelity level is unspecified.
+  SYMBOL_FIDELITY_FULL = 1;        // The symbol fidelity is at its fullest level of detail.
+}
+
+// Describes the mapping from a binary to its original source code. These are stored in a lookup table in a Profile. These are referenced by index from other messages.
+message Mapping {
+  // Address at which the binary (or DLL) is loaded into memory.
+  uint64 memory_start = 1;
+  // The limit of the address range occupied by this mapping.
+  uint64 memory_limit = 2;
+  // Offset in the binary that corresponds to the first mapped address.
+  uint64 file_offset = 3;
+  // The object this entry is loaded from.  This can be a filename on
+  // disk for the main binary and shared libraries, or virtual
+  // abstractions like "[vdso]". Index into string table
+  uint32 filename_index = 4;
+  // A string that uniquely identifies a particular program version
+  // with high probability. E.g., for binaries generated by GNU tools,
+  // it could be the contents of the .note.gnu.build-id field. Index into string table
+  uint32 build_id_index = 5;
+
+  SymbolFidelity symbolic_info = 6;
+
+  // Reference to an attribute set from the Profile's attribute set table.
+  uint32 attribute_set_index = 7;
+}
+
+// Describes function and line table debug information. These are stored in a lookup table in a Profile. These are referenced by index from other messages.
+message Location {
+  // The id of the corresponding profile.Mapping for this location.
+  // It can be unset if the mapping is unknown or not applicable for
+  // this profile type.
+  uint32 mapping_index = 1;
+  // The instruction address for this location, if available.  It
+  // should be within [Mapping.memory_start...Mapping.memory_limit]
+  // for the corresponding mapping. A non-leaf address may be in the
+  // middle of a call instruction. It is up to display tools to find
+  // the beginning of the instruction if necessary.
+  uint64 address = 2;
+  // Multiple line indicates this location has inlined functions,
+  // where the last entry represents the caller into which the
+  // preceding entries were inlined.
+  //
+  // E.g., if memcpy() is inlined into printf:
+  //    line[0].function_name == "memcpy"
+  //    line[1].function_name == "printf"
+  repeated Line line = 3;
+
+  // Reference to an attribute set from the Profile's attribute set table.
+  uint32 attribute_set_index = 4;
+}
+
+// Represents a line in a source file. These are embedded within a Location message.
+message Line {
+  // The id of the corresponding profile.Function for this line.
+  uint32 function_index = 1;
+  // Line number in source code.
+  uint32 line = 2;
```

Comment:
Is this optional? What if we don't have debug information to map to source lines, or the code is auto-generated and doesn't have source lines?